src/NotificationCenter/Notifications.hs

```diff
@@ -441,18 +445,57 @@ removeNotiFromDistList tState id = do
     return False
   return ()
 
+-- | Adjusts the position of all displayed notifications so they follow standardized placement rules.
+readjustNotificationPositions :: Config -> TVar NotifyState -> IO ()
+readjustNotificationPositions config tState = do
+  sortedDisplayedPopups <- sortOn _dNotiTop . filter (not . _dHasCustomPosition) . notiDisplayingList <$> readTVarIO tState
+  newDisplayedPopups <- pushNotificationsUp config sortedDisplayedPopups
+  atomically $ modifyTVar' tState $ \state ->
```

Comment:
I could be wrong, but this could loose notifications. Consider this scenario:
- You read display notis out in line 451
- then another process might kick in and adds a notification
- You overwrite displayNoti List w/o new noti (as it was not present before)

It probably has all to be atomic